### PR TITLE
fix(driver): properly support more ia32 syscalls "manual" translation to x86_64 ones

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area driver-kmod
/area driver-bpf
/area driver-modern-bpf
/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Follow up to #1196 : this PR leverages https://github.com/falcosecurity/syscalls-bumper/pull/19 to properly translate more "ia32 only" syscalls to x86_64 ones, for compatible syscalls.
With this PR, all the  non-generic syscalls that were previously captured by the kmod (before #1196) with the previous kmod-only ia32 support will be now captured by all 3 drivers.
This means:
* `sendfile64, getgid32, fcntl64, setuid32, setresuid32, getegid32, getresgid32, getresuid32, setresgid32, geteuid32, setgid32` will now be translated to `sendfile` , `getgid`, `fcntl`, and so on, since they already generated the same event as the 64bit version
* `stat64, lstat64, fstat64` will now be translated to `stat`, `lstat` and `fstat` (just like we did for `mmap2`, since the generated events are compatible)
 * `umount` will be mapped to `umount2` since it generates a compatible event; `PPME_SYSCALL_UMOUNT2_E` `flags` parameter will be empty when coming from ia32 since `umount` has no such param
 * `_llseek` will be mapped to `lseek` since it generates a compatible event
* `ugetrlimit` will be mapped to `getrlimit` since it generates a compatible event

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Following on https://github.com/falcosecurity/libs/pull/1196#issuecomment-1757774971 comment, we had:
* 16 specific events/syscalls that previously the kmod-only impl was able to capture, and the new ia32 common impl was not
* 19 generic syscalls that previously the kmod-only impl was able to capture and the new ia32 common impl was not
* ~35 syscalls that were never captured

The latter ~35 syscalls are useless since they were not captured before.
With this PR, we will cover the 16 specific syscalls that were missing in the new impl:
```
[221] = 72, // NOTE: syscall fcntl64 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [196] = 6, // NOTE: syscall lstat64 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [211] = 120, // NOTE: syscall getresgid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [208] = 117, // NOTE: syscall setresuid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [200] = 104, // NOTE: syscall getgid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [22] = 166, // NOTE: syscall umount unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [201] = 107, // NOTE: syscall geteuid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [209] = 118, // NOTE: syscall getresuid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [192] = 9, // NOTE: syscall mmap2 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [213] = 105, // NOTE: syscall setuid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [197] = 5, // NOTE: syscall fstat64 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [195] = 4, // NOTE: syscall stat64 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [202] = 108, // NOTE: syscall getegid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [214] = 106, // NOTE: syscall setgid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [239] = 40, // NOTE: syscall sendfile64 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
        [210] = 119, // NOTE: syscall setresgid32 unmapped on x86_64, forcefully mapped to compatible syscall. See syscalls-bumper bumpIA32to64Map() call.
```
We still miss 3 specific syscalls:
```
* [140] = -1, // ia32 only: _llseek 
* [199] = -1, // ia32 only: getuid32
* [191] = -1, // ia32 only: ugetrlimit
```

And we still miss the following 19 generic syscalls, ie:
```
* [48] = -1, // ia32 only: signal
* [69] = -1, // ia32 only: ssetmask
* [89] = -1, // ia32 only: readdir
* [34] = -1, // ia32 only: nice
* [73] = -1, // ia32 only: sigpending
* [68] = -1, // ia32 only: sgetmask
* [117] = -1, // ia32 only: ipc
* [7] = -1, // ia32 only: waitpid
* [67] = -1, // ia32 only: sigaction
* [300] = -1, // ia32 only: fstatat64
* [109] = -1, // ia32 only: olduname
* [112] = -1, // ia32 only: idle
* [72] = -1, // ia32 only: sigsuspend
* [269] = -1, // ia32 only: fstatfs64
* [268] = -1, // ia32 only: statfs64
* [126] = -1, // ia32 only: sigprocmask
* [142] = -1, // ia32 only: _newselect
* [25] = -1, // ia32 only: stime
* [119] = -1, // ia32 only: sigreturn
```

**Comment by FedeDP** [comment]
> We still miss 3 specific syscalls:

Gonna fix these ones if possible!

File: driver/syscall_ia32_64_map.c
Comment (FedeDP) [inline driver/syscall_ia32_64_map.c:1]: Most of the changes come from the new sorting of this file.

**Review by Andreagit97 — APPROVED** [review]
/approve
Thank you for this!

**Comment by Andreagit97** [comment]
Again the SCHEMA VERSION here is another false positive
/unhold

**Review by jasondellaluce — APPROVED** [review]

**Pull request merged**